components/ai_chat/core/browser/ai_chat_service.cc

```diff
@@ -675,6 +678,7 @@ void AIChatService::MaybeUnloadConversation(
     return;
```

Comment:
This part shouldn't happen if the conversation is temporary, right? If there isn't a client connected then there's no way to load the conversation again, so we can delete the in-memory state.

Reply:
I'm fine with starting with more strict policy as you suggested here, and make changes if others disagree with the experience, will push a commit for this.

Reply:
https://github.com/brave/brave-core/pull/29187/commits/3a6a1bbe47c58af67efe631e3a1cd531439e269e